### PR TITLE
[WEB-1159] Patient access management

### PR DESCRIPTION
**Pull request by cbwebdevelopment**
See [WEB-1159]

Related PR: https://github.com/tidepool-org/platform-client/pull/123

[WEB-1159]: https://tidepool.atlassian.net/browse/WEB-1159

### CODE REVIEW

**Review by krystophv — APPROVED** [review]
A handful of comments and suggestions, but nothing major. Great work on this,

LGTM 👍🏼

File: app/pages/share/AccessManagement.js [hunk 1]
```diff
@@ -0,0 +1,645 @@
...
+        body,
+        submitText,
+      })
+    }
+  }, [selectedSharedAccount]);
+
+  function handleUploadPermissionsToggle(member) {
+    trackMetric(`upload permission turned ${member.uploadPermission ? 'off' : 'on'}`);
+
+    const permissions = {
```
[the remark this answers is not in the document]
Comment (cbwebdevelopment) [reply app/pages/share/AccessManagement.js:306]: Yes, good call.  I just lifted this from the current share page logic.  Should definitely have looked at this with fresh eyes.

File: app/redux/actions/async.js [hunk 2]
```diff
@@ -2014,6 +2094,30 @@ export function deleteClinicianInvite(api, clinicId, inviteId) {
...
+ * Send Clinic Invite Action Creator
+ *
+ * @param {Object} api - an instance of the API wrapper
+ * @param {String} shareCode - share code of the clinic to invite
+ * @param {Object} permissions - permissions to be given
+ * @param {String} patientId - id of the patient sending the invite
+ */
+ export function sendClinicInvite(api, shareCode, permissions, patientId) {
+  return (dispatch) => {
+    dispatch(sync.sendInviteRequest());
```
[the remark this answers is not in the document]
Comment (cbwebdevelopment) [reply app/redux/actions/async.js:2107]: I went back and forth on this myself, but I think I agree.  Will add a separate action and push shortly

**Comment by cbwebdevelopment** [comment]
Just as a heads-up, @krystophv, I've incorporated all of your feedback.  Since this was already approved, I'll go ahead and merge it once CI checks pass, but if you'd like to review the changes, they're 'mostly' in this commit:
https://github.com/tidepool-org/blip/pull/913/commits/b263496eb04873eb787f9d1ad1f43bb5ce6249fc

The bulk of it is just the addition of the separate invite action for the clinic and it's tests.

**Comment by krystophv** [comment]
Yup, I've taken a look over the new commits. Everything LGTM 👍🏼

**Pull request merged**